src/backend/trace.rs

```diff
@@ -468,3 +468,229 @@ impl TraceData {
+                        dest_port,
+                        ttl,
+                        round,
+                        sent,
+                        flags,
+                    ))),
+                    "c" => {
+                        let host = IpAddr::from_str(values[3])?;
+                        let duration = Duration::from_millis(u64::from_str(values[2])?);
+                        let received = sent.add(duration);
+                        let icmp_packet_type = IcmpPacketType::NotApplicable;
+                        Ok(ProbeState::Complete(
+                            Probe::new(
+                                sequence,
+                                TraceId(0),
+                                src_port,
+                                dest_port,
+                                ttl,
+                                round,
+                                sent,
+                                flags,
+                            )
+                            .complete(
+                                host,
+                                received,
+                                icmp_packet_type,
+                                None,
+                            ),
+                        ))
+                    }
+                    _ => Err(anyhow!("unknown probe state")),
+                }?;
+                Ok(Self(state))
+            } else {
+                Err(anyhow!("failed to parse {}", value))
+            }
+        }
+    }
+
+    /// A helper struct so wwe may inject the round into the probes.
+    struct ProbeRound(ProbeData, Round);
+
+    impl From<ProbeRound> for ProbeState {
+        fn from(value: ProbeRound) -> Self {
+            let probe_data = value.0;
+            let round = value.1;
+            match probe_data.0 {
+                Self::NotSent => Self::NotSent,
+                Self::Skipped => Self::Skipped,
+                Self::Awaited(awaited) => Self::Awaited(Probe { round, ..awaited }),
+                Self::Complete(completed) => Self::Complete(ProbeComplete { round, ..completed }),
+            }
+        }
+    }
+
+    /// The expected outcome.
+    ///
+    /// TODO include flows here.
+    #[derive(Deserialize, Debug, Clone)]
+    #[serde(deny_unknown_fields)]
+    struct Expected {
+        /// The expected outcome per hop.
+        hops: Vec<HopData>,
+    }
+
+    /// The expected outcome for a single hop.
+    #[derive(Deserialize, Debug, Clone)]
+    #[serde(deny_unknown_fields)]
+    struct HopData {
+        ttl: u8,
+        total_sent: usize,
+        total_recv: usize,
+        loss_pct: f64,
+        last_ms: Option<f64>,
+        best_ms: Option<f64>,
+        worst_ms: Option<f64>,
+        avg_ms: f64,
+        jitter: Option<f64>,
+        javg: f64,
+        jmax: Option<f64>,
+        jinta: f64,
+        addrs: HashMap<IpAddr, usize>,
+        samples: Option<Vec<f64>>,
+        last_src: u16,
+        last_dest: u16,
+        last_sequence: u16,
```

Comment:
@trkelly23 I added all remaining fields we need to test.  I also adjusted which are `Option<T>` to match the public interface.  I only updated `test_scenario.yaml` to match for now, the other scenarios will need to be updated as well if we're happy with this approach.